catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/RestoreCommand.java

```diff
@@ -0,0 +1,319 @@
+            + " the status of a given restore. When requesting a restore status, --restoreStatus and --requestId"
+            + " are both required options."
+  )
+  @VisibleForTesting
+  protected String requestId;
+
+  @Option(
+    name = "-a",
+    aliases = {"--asyncRestore"},
+    description = "Perform an asynchronous restore."
+  )
+  @VisibleForTesting
+  protected boolean asyncRestore;
+
+  @Option(
+    name = "-f",
+    aliases = {"--force"},
+    description = "Forces the restore. Will delete the collection if it already exists"
+  )
+  @VisibleForTesting
+  protected boolean force = false;
+
+  @Override
+  public Object execute() throws Exception {
+    if (isSystemConfiguredWithSolrCloud()) {
+      performSolrCloudRestore();
+    } else {
+      performSingleNodeSolrRestore();
+    }
+
+    return null;
+  }
+
+  private void performSingleNodeSolrRestore() throws URISyntaxException {
+    String restoreUrl = getReplicationUrl(coreName);
+
+    try {
+      createSolrCore();
+
+      httpBuilder = new org.codice.solr.factory.impl.HttpClientBuilder(encryptionService);
+      URIBuilder uriBuilder = new URIBuilder(restoreUrl);
+      uriBuilder.addParameter("command", "restore");
+
+      if (StringUtils.isNotBlank(backupLocation)) {
+        uriBuilder.addParameter("location", backupLocation);
+      }
+
+      URI restoreUri = uriBuilder.build();
+      LOGGER.debug("Sending request to {}", restoreUri);
+
+      printResponse(sendGetRequest(restoreUri));
+    } catch (IOException | SolrServerException e) {
+      LOGGER.info("Unable to perform single node Solr restore, core: {}", coreName, e);
+    }
+  }
+
+  private void performSolrCloudRestore() throws IOException {
+    SolrClient client = null;
+    try {
+      client = getCloudSolrClient();
+      if (status) {
+        verifyStatusInput();
+        printStatus(client, requestId);
+      } else {
+        verifyRestoreInput();
+        performSolrCloudRestore(client);
+      }
+    } finally {
+      shutdown(client);
+    }
+  }
+
+  private void printResponse(@Nullable HttpResponse response) {
+    if (response != null) {
+      StatusLine statusLine = response.getStatusLine();
+      if (statusLine.getStatusCode() == HttpStatus.SC_OK) {
+        printSuccessMessage(String.format("%nRestore of [%s] complete.%n", coreName));
+      } else {
+        printErrorMessage(String.format("Error restoring Solr core: [%s]", coreName));
+        printErrorMessage(
+            String.format(
+                "Restore command failed due to: %d - %s",
+                statusLine.getStatusCode(), statusLine.getReasonPhrase()));
+        printErrorMessage(String.format("Request: %s", getReplicationUrl(coreName)));
+      }
+    }
+  }
+
+  private boolean restore(
+      SolrClient client, String collection, String backupLocation, String backupName)
+      throws IOException, SolrServerException {
+    if (canRestore(client, collection)) {
```

Comment:
✏️ If you negate the condition in this `if` statement, you can short cirucit and won't need an `else`:
```
if (!canRestore(client, collection)) {
      LOGGER.warn("Unable to restore collection {}", collection);
      return false;
}
// rest of method
```